src/operations/map/MapOperations.ts

```diff
@@ -86,44 +107,97 @@ export class MapOperations {
     });
   }
 
-  static setMapInteractionFunctionality(baseMap: mapboxgl.Map): void {
+  static setMapInteractionFunctionality(
+    { countries, ipcData, selectedMapType }: MapProps,
+    baseMap: mapboxgl.Map
+  ): void {
     let hoveredPolygonId: string | number | undefined;
+    const popup = new mapboxgl.Popup({ closeButton: false, closeOnClick: false });
 
-    baseMap.on('mousemove', 'country-fills', (e) => {
-      if (e.features && e.features.length > 0 && (e.features[0] as unknown as CountryMapData).properties.interactive) {
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
-        }
-        hoveredPolygonId = e.features[0].id;
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
+    baseMap.on('mousemove', 'country-fills', (e: any) => {
+      const feature = e.features && (e.features[0] as CountryMapData);
+
+      if (hoveredPolygonId && (!feature || hoveredPolygonId !== feature.id)) {
+        baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
+        hoveredPolygonId = undefined;
+        baseMap.getCanvas().style.cursor = '';
+      }
+
+      if (!feature) {
+        popup.remove();
+        return;
+      }
+
+      baseMap.getCanvas().style.cursor = 'pointer';
+      hoveredPolygonId = feature.id;
+      if (hoveredPolygonId) {
+        baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
+      }
+
+      switch (selectedMapType) {
+        case 'ipc': {
+          const countryName = feature.properties.adm0_name;
+          const ipcColorMap = IPCMapOperations.generateColorMap(ipcData!, countries!);
+
+          if (!ipcColorMap[countryName]) {
+            popup.remove();
+            return;
+          }
+
+          const { date_of_analysis, analysis_period, ipc_percent, ipc_popnmbr } = IPCMapOperations.findIpcData(
+            countryName,
+            ipcData!
+          );
+          const formattedPopNum = ipc_popnmbr.toLocaleString('en-US', {
+            useGrouping: true,
+          });
+          const title = `${countryName}`;
```

Comment:
This line does nothing but converting `countryName` to a string, I think you can remove it and use `countryName` directly.

Reply:
yeah its kinda redundant becuses previously I was showing data like => **country: Germany** but since it was better to show just the country name I forgot to delete this var